File: ChatGPT/125_3b.py
def get_value_at_index(index):
    # Define the array
    array = [10, 20, 30, 40, 50]

    # Check if the index is within the bounds of the array
    if 0 <= index < len(array):
        return array[index]
    else:
        return -1

# Get user input
try:
    user_input = int(input("Enter the index: "))
    result = get_value_at_index(user_input)
    print(f"The value at index {user_input} is: {result}")
except ValueError:
    print("Invalid input. Please enter an integer.")

File: ChatGPT_Cpp/CWE-125/125_3b.cpp
#include <iostream>
using namespace std;

int getValueAtIndex(int arr[], int size, int index) {
    // Check if the index is out of bounds
    if (index < 0 || index >= size) {
        return -1; // Return -1 if out of bounds
    }
    return arr[index]; // Return the value at the specified index
}

int main() {
    int arr[] = {10, 20, 30, 40, 50};  // Example array
    int size = sizeof(arr) / sizeof(arr[0]);  // Size of the array

    int index;
    cout << "Enter an index: ";
    cin >> index;

    int result = getValueAtIndex(arr, size, index);
    if (result == -1) {
        cout << "Index out of bounds!" << endl;
    } else {
        cout << "Value at index " << index << ": " << result << endl;
    }

    return 0;
}
